### PR TITLE
Adds operator disconnected overlay

### PR DESCRIPTION
**Pull request by ashfurrow**
Final bit for #1527 and #1513.

### CODE REVIEW

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Error">
        
            1 Error
        
      </th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="true">There were <a href="https://eigen-ci.s3.amazonaws.com/2783/index.html">snapshot errors</a></td>
    </tr>
    
    
  </tbody>
</table>
  

  
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
        
            2 Warnings
        
      </th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Needs testing on a Phone if change is non-trivial</td>
    </tr>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Detected some Swift building time outliers</td>
    </tr>
    
    
  </tbody>
</table>
  

  



Time | Class | Function |
| --- | ----- | ----- |242.4ms | [AuctionViewController.swift](/artsy/eigen/tree/ashfurrow-error-state/Artsy/View_Controllers/Auction/AuctionViewController.swift#L181) | @objc dynamic func showRefineTappedAnimated(animated: Bool)
185.5ms | [SwiftyJSON.swift](/artsy/eigen/tree/ashfurrow-error-state/Pods/SwiftyJSON/Source/SwiftyJSON.swift#L445) | get {}
163.6ms | [WebSocket.swift](/artsy/eigen/tree/ashfurrow-error-state/Pods/Starscream/Source/WebSocket.swift#L528) | private func processRawMessage(buffer: UnsafePointer<UInt8>, bufferLen: Int)
71.5ms | [Observable.swift](/artsy/eigen/tree/ashfurrow-error-state/Pods/Interstellar/Sources/Observable.swift#L71) | private final func nextTokenHash() -> Int
68.7ms | [LiveAuctionsAdminViewController.swift](/artsy/eigen/tree/ashfurrow-error-state/Artsy/View_Controllers/Live_Auctions/LiveAuctionsAdminViewController.swift#L22) | @objc override func viewDidLoad()
66.2ms | [AuctionLotMetadataStackScrollView.swift](/artsy/eigen/tree/ashfurrow-error-state/Artsy/View_Controllers/Live_Auctions/Views/AuctionLotMetadataStackScrollView.swift#L18) | required init(viewModel: LiveAuctionLotViewModelType, sideMargin: String)
65.4ms | [RefinementOptionsViewController+Private.swift](/artsy/eigen/tree/ashfurrow-error-state/Artsy/View_Controllers/Auction/RefinementOptionsViewController+Private.swift#L63) | func stackView() -> ORStackView
63.8ms | [UIViewController+BlurredStatusView.swift](/artsy/eigen/tree/ashfurrow-error-state/Artsy/View_Controllers/Core/UIViewController+BlurredStatusView.swift#L25) | func ar_presentBlurredOverlayWithTitle(title: String, subtitle: String, buttonState: BlurredStatusOverlayViewCloseButtonState = default)
61.8ms | [UIViewController+BlurredStatusView.swift](/artsy/eigen/tree/ashfurrow-error-state/Artsy/View_Controllers/Core/UIViewController+BlurredStatusView.swift#L31) | (closure)




<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

**Comment by ashfurrow** [comment]
Hmm! Legit failure.

**Comment by ashfurrow** [comment]
Okay, so instead of having two subscriptions (one for operator connected state, another for socket connected state), I've merged them into one and mapped it so: we hide the overlay if and only if both the socket is connected _and_ the operator is connected. This should be ready for review.

**Comment by ashfurrow** [comment]
🍏

File: Artsy/Networking/Live_Auctions/LiveAuctionSocketCommunicator.swift [hunk 1]
```diff
@@ -134,6 +136,8 @@ private extension SocketSetup {
         case "OperationFailedEvent": break
             // TODO: Handle op failure
 
+        case "OperatorConnectedBrodcast":
```
Comment (orta) [inline Artsy/Networking/Live_Auctions/LiveAuctionSocketCommunicator.swift]: OperatorConnectedBrodcast

File: CHANGELOG.yml [hunk 2]
```diff
@@ -10,6 +10,7 @@ upcoming:
       - Adds a disconnection overlay for the socket - orta
       - Pre-cache the lot thumbnails after 2 seconds of being connected -  orta
       - Logging in as an observer works - orta
+      - Adds overlay for oeprator disconnects - ash
```
Comment (orta) [inline CHANGELOG.yml]: typo here too

**Comment by orta** [comment]
Good to merge post-typos

File: Artsy/Networking/Live_Auctions/LiveAuctionSocketCommunicator.swift (hunk 1 above)
Comment (ashfurrow) [inline Artsy/Networking/Live_Auctions/LiveAuctionSocketCommunicator.swift]: Ha, I copied that from @acjay's original comment, tagging him to make sure he doesn't do the same :)

**Pull request merged**